packages/cardano-services-client/src/SharedWalletProvider/BlockfrostSharedWalletProvider.ts

```diff
@@ -0,0 +1,26 @@
+import * as Crypto from '@cardano-sdk/crypto';
+import { BlockfrostClient, BlockfrostProvider } from '../blockfrost';
+import { Logger } from 'ts-log';
+import { MULTI_SIG_LABEL, MultiSigTransaction, SharedWalletProvider } from '@cardano-sdk/core';
+
+export class BlockfrostSharedWalletProvider extends BlockfrostProvider implements SharedWalletProvider {
+  constructor(client: BlockfrostClient, logger: Logger) {
+    super(client, logger);
+  }
+
+  async discoverWallets(pubKey: Crypto.Ed25519KeyHashHex): Promise<MultiSigTransaction[]> {
+    let page = 1;
+    const pageSize = 100;
+    let allTransactions: MultiSigTransaction[] = [];
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const response = await this.request<MultiSigTransaction[]>(`metadata/txs/labels/${MULTI_SIG_LABEL}?page=${page}`);
+      allTransactions = [...allTransactions, ...response];
+      if (response.length < pageSize) break;
+      page++;
+    }
```

Comment:
We have a util that does this. Search for usages for examples.
```suggestion
const allTransactions = await fetchSequentially(...)
```